api/controller.py

```diff
@@ -327,16 +327,17 @@ def setup_adobe_vendor_id(self, _db, library):
             ' worry about.'
         )
 
+        new_adobe_vendor_id = None
         if adobe:
             # Relatively few libraries will have this setup.
             vendor_id = adobe.username
             node_value = adobe.password
             if vendor_id and node_value:
-                if self.adobe_vendor_id:
+                if new_adobe_vendor_id:
```

Comment:
Sometimes `self.adobe_vendor_id` wasn't set and this was crashing. I couldn't find the reason for this, but the new idiom is safer and avoids rapid changes to `self.adobe_vendor_id` while the configuration is loading.